lib/basedriver/commands/session.js

```diff
@@ -15,34 +15,27 @@ commands.createSession = async function createSession (jsonwpDesiredCapabilities
 
   log.debug();
 
-  // Determine weather we should use jsonwpDesiredCapabilities or w3cCapabilities to get caps from
-  let caps;
-  if (w3cCapabilities) {
-    this.setProtocolW3C();
-    if (jsonwpDesiredCapabilities) {
-      log.debug(`W3C capabilities and MJSONWP desired capabilities were provided`);
-    }
+  if (!w3cCapabilities) {
+    throw new errors.SessionNotCreatedError('Appium only supports W3C capabilitites');
```

Comment:
`Appium only supports W3C-style capability objects. Your client is sending an older capabilities format. Please update your client library.`